File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/analysis/network/NetworkNode.java
package io.sch.historyscan.domain.contexts.analysis.network;

import java.util.Comparator;
import java.util.List;

public record NetworkNode(
        String name,
        String path,
        String parentPath,
        long currentNbLines,
        long score,
        List<NetworkLink> links
) implements Comparable<NetworkNode> {

    @Override
    public int compareTo(NetworkNode o) {
        return Comparator
                .comparing(NetworkNode::score, Comparator.reverseOrder())
                .thenComparing(NetworkNode::name)
                .compare(this, o);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/network/NetworkNode.kt
package io.sch.historyscan.domain.contexts.analysis.network

class NetworkNode(
    val name: String,
    val path: String,
    val parentPath: String,
    val currentNbLines: Long,
    val score: Long,
    val links: List<NetworkLink>
) : Comparable<NetworkNode> {
    override fun compareTo(other: NetworkNode): Int {
        return Comparator
            .comparing(NetworkNode::score, Comparator.reverseOrder())
            .thenComparing(NetworkNode::name)
            .compare(this, other)
    }
}
